### PR TITLE
feat(grouping):#MVS-51 delete audience from grouping

### PR DESCRIPTION
**Pull request by vava56754**
## Describe your changes
Suppression d'une classe ou d'un regroupement appartenant au regroupement concerné.
https://confluence.support-ent.fr/display/MTVS/Contrat+d%27API
## Checklist tests
- [x] supression d'une classe ou d'un groupe d'un regroupement
## Issue ticket number and link
MVS-51: https://jira.support-ent.fr/browse/MVS-51
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [x] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by aymericdevriesere — APPROVED** [review]

File: src/main/java/fr/openent/viescolaire/service/impl/DefaultGroupingService.java [hunk 1]
```diff
@@ -159,4 +159,22 @@ public Future<JsonObject> deleteGrouping(String groupingId) {
...
+        Promise<JsonObject> promise = Promise.promise();
+        JsonArray values = new JsonArray();
+        String query = "DELETE FROM " + TABLE_REL + " WHERE " + TABLE_REL + ".grouping_id = ? AND " + TABLE_REL + ".student_division_id = ?";
+        values.add(groupingId);
+        values.add(studentDivisionId);
+        Sql.getInstance().prepared(query,values, SqlResult.validUniqueResultHandler(res -> {
+            if (res.isRight())
+                promise.complete(new JsonObject().put(Field.STATUS, Field.OK));
+            else {
+                String messageToFormat = "[vie-scolaire@%s::deleteGrouping] Error while deleting class or group to the grouping : %s";
```
Comment (flmartineau) [inline src/main/java/fr/openent/viescolaire/service/impl/DefaultGroupingService.java:174]: [Viescolaire@%s::deleteGroupingAudience]

File: src/main/java/fr/openent/viescolaire/service/impl/DefaultGroupingService.java [hunk 2]
```diff
@@ -159,4 +159,22 @@ public Future<JsonObject> deleteGrouping(String groupingId) {
         }));
         return promise.future();
     }
+
+    @Override
+    public Future<JsonObject> deleteGroupingAudience(String groupingId, String studentDivisionId) {
+        Promise<JsonObject> promise = Promise.promise();
+        JsonArray values = new JsonArray();
+        String query = "DELETE FROM " + TABLE_REL + " WHERE " + TABLE_REL + ".grouping_id = ? AND " + TABLE_REL + ".student_division_id = ?";
```
Comment (qperie) [inline src/main/java/fr/openent/viescolaire/service/impl/DefaultGroupingService.java:167]: Simplifiable par : "DELETE FROM " + TABLE_REL + " WHERE grouping_id = ? AND student_division_id = ?";

File: src/test/java/fr/openent/viescolaire/service/impl/DefaultGroupingServiceTest.java [hunk 3]
```diff
@@ -126,4 +126,23 @@ public void TestDeleteGrouping(TestContext ctx) {
         defaultGroupingService.deleteGrouping(groupingTestId);
     }
 
+    @Test
+    public void TestDeleteGroupingAudience(TestContext ctx) {
+        Async async = ctx.async();
+        String queryExpected = "DELETE FROM " + tableRel + " WHERE " + tableRel + ".grouping_id = ? AND " + tableRel + ".student_division_id = ?";
```
Comment (qperie) [inline src/test/java/fr/openent/viescolaire/service/impl/DefaultGroupingServiceTest.java:132]: Simplifiable par : "DELETE FROM " + tableRel + " WHERE grouping_id = ? AND student_division_id = ?";

File: src/main/java/fr/openent/viescolaire/service/impl/DefaultGroupingService.java (hunk 2 above)
Comment (aymericdevriesere) [reply src/main/java/fr/openent/viescolaire/service/impl/DefaultGroupingService.java:167]: oui mais en soit explicite

File: src/test/java/fr/openent/viescolaire/service/impl/DefaultGroupingServiceTest.java (hunk 3 above)
Comment (qperie) [reply src/test/java/fr/openent/viescolaire/service/impl/DefaultGroupingServiceTest.java:132]: Magic string pour les tableRel ?

**Review by alicedraillard — APPROVED** [review]

**Review by flmartineau — APPROVED** [review]

**Pull request merged**